Source/WebKitLegacy/mac/WebView/WebView.mm

```diff
@@ -8406,26 +8406,32 @@ + (void)_setCacheModel:(WebCacheModel)cacheModel
     if (s_didSetCacheModel && cacheModel == s_cacheModel)
         return;
 
-    auto nsurlCacheDirectory = adoptCF(_CFURLCacheCopyCacheDirectory([[NSURLCache sharedURLCache] _CFURLCache]));
-    if (!nsurlCacheDirectory)
-        nsurlCacheDirectory = (__bridge CFStringRef)NSHomeDirectory();
+    RetainPtr<CFURLCacheRef> cfurlCache = adoptCF(CFURLCacheCopySharedURLCache());
+    auto cfurlCacheDirectory = adoptCF(_CFURLCacheCopyCacheDirectory(cfurlCache.get()));
+    if (!cfurlCacheDirectory) {
+#if PLATFORM(IOS_FAMILY)
+        CFURLRef url = CFCopyHomeDirectoryURL();
+        cfurlCacheDirectory = CFURLGetString(url);
+        CFRelease(url);
+#else
+        cfurlCacheDirectory = (__bridge CFStringRef)NSHomeDirectory();
+#endif
+    }
```

Comment:
There is a whole lot of subtle behavior differences between functions used here. While I don't have a specific problematic scenario to pinpoint, it is more likely than not that we'd break something with this. And given that this is legacy code that is not performance sensitive at all, I'd recommend leaving it as is.